File: src/RentCars.Domain/Users/NameOfUser.cs
namespace RentCars.Domain.Users;

public class NameOfUser
{
    public Guid Id { get; }
    public String Name { get; }
    public String Login { get; }

    public NameOfUser(Guid id, String name, String login)
    {
        Id = id;
        Name = name;
        Login = login;
    }
}

File: API/api.client/src/domain/users/nameOfUser.ts
import { RoleType } from "./enums/roleType";

export class NameOfUser {
    constructor(
        public id: string,
        public name: string,
        public login: string,
    ) { }
}

export function mapToNameOfUser(data: any) {
    return new NameOfUser(
        data.id,
        data.name,
        data.login,
    );
}